### PR TITLE
Upgrade UI deps

### LINKED ISSUE #3162: Failing tests after upgrading UI dependencies
TBD

### Description:

Last time we tried it, we got an error, that, presumably, has something to do with a transitive dependency.

### Steps to reproduce the issue:

1. `yarn test` -> ok
2. `yarn upgrade`
3. `yarn test` -> nok

### Describe the results you received:

Tests failing in `AppView` related views when detecting the cluster being passed
```
    @@ -1,9 +1,9 @@
      Array [
        ResourceRef {
          "apiVersion": "apps/v1beta1",
    -     "cluster": "cluster-1",
    +     "cluster": undefined,
    > 403 |     expect(applicationStatus.prop("statefulsetRefs")).toEqual([
 ```

### Describe the results you expected:

Tests passing OK

### Additional information you deem important (e.g. issue happens only occasionally):

N/A

### Version of Helm, Kubeapps and Kubernetes:

N/A (dev)

### PR DESCRIPTION
**Pull request by antgamdia**
### Description of the change

This  PR simply upgrades the UI deps, moves (and cleans up) the types deps to devOnly ones and performs minor changes related to the types (ie adding missing fields)

### Benefits

We will be able to safely upgrade our UI deps again.

### Possible drawbacks

Like in any other upgrade, perhaps we are missing something hidden behind the unit, e2e and manual tests. 
### Applicable issues

  - fixes #3162

### Additional information

Thanks @dlaloue-vmware  for jumping in and suggesting the isomorphic-form-data dep.

### CODE REVIEW

File: dashboard/package.json [hunk 1]
```diff
@@ -10,24 +10,6 @@
...
-    "@types/qs": "^6.9.7",
-    "@types/react": "^17.0.15",
-    "@types/react-copy-to-clipboard": "^5.0.1",
-    "@types/react-jsonschema-form": "^1.7.6",
-    "@types/react-router-dom": "^5.1.8",
-    "@types/react-router-hash-link": "^2.4.1",
-    "@types/react-tabs": "^2.3.3",
-    "@types/react-transition-group": "^4.4.2",
-    "@types/semver": "^7.3.8",
-    "@types/ws": "^7.4.7",
```
Comment (antgamdia) [inline dashboard/package.json:30]: Moved to dev deps. Also, I removed all of them and just installed those ones Typescript was complaining (this way we reduce the types dependencies (as many of them are including also the definition of the types)

File: dashboard/src/components/AppList/AppList.test.tsx [hunk 2]
```diff
@@ -57,15 +56,25 @@ context("when changing props", () => {
     jest.spyOn(qs, "parse").mockReturnValue({
       q: "foo",
     });
-    const wrapper = mountWrapper(defaultStore, <AppList />);
+    const wrapper = mountWrapper(
+      defaultStore,
+      <MemoryRouter initialEntries={["/foo?q=foo"]}>
+        <AppList />
+      </MemoryRouter>,
```
Comment (antgamdia) [inline dashboard/src/components/AppList/AppList.test.tsx:63]: The main change here is replacing the jest.spyOn with the approach suggested in the official website: https://reactrouter.com/web/guides/testing

File: dashboard/src/components/Header/ContextSelector.test.tsx [hunk 3]
```diff
@@ -201,18 +198,17 @@ it("changes the location with the new namespace", () => {
...
 it("changes the location with the new cluster and namespace", () => {
-  const push = jest.fn();
-  spyOnUseHistory = jest.spyOn(ReactRouter, "useHistory").mockReturnValue({ push } as any);
-  spyOnUseLocation = jest.spyOn(ReactRouter, "useLocation").mockReturnValue({
-    pathname: "/c/default-cluster/ns/ns-bar/catalog",
-    search: "",
-    state: "",
-  } as any);
-  const wrapper = mountWrapper(defaultStore, <ContextSelector />);
+  const history = createMemoryHistory({ initialEntries: ["/c/default-cluster/ns/ns-bar/catalog"] });
```
Comment (antgamdia) [inline dashboard/src/components/Header/ContextSelector.test.tsx:205]: An alternative way to the <MemoryRouter initialEntries=......> which allows using the variable to perform the checks

File: dashboard/src/components/js/Checkbox/Checkbox.test.js [hunk 4]
```diff
@@ -3,6 +3,7 @@ import { shallow } from "enzyme";
 import Checkbox from ".";
 
 const defaultProps = {
+  id: "test",
```
Comment (antgamdia) [inline dashboard/src/components/js/Checkbox/Checkbox.test.js:6]: A test that was generating an annoying warning for long time ago :P

File: dashboard/src/containers/AppUpgradeContainer/AppUpgradeContainer.test.tsx [hunk 5]
```diff
@@ -1,15 +1,18 @@
...
+import { Location } from "history";
 import configureMockStore from "redux-mock-store";
 import thunk from "redux-thunk";
 import AppUpgrade from "../../components/AppUpgrade";
 import Upgrade from "./AppUpgradeContainer";
 
 const mockStore = configureMockStore([thunk]);
 
-const emptyLocation = {
+const emptyLocation: Location = {
```
Comment (antgamdia) [inline dashboard/src/containers/AppUpgradeContainer/AppUpgradeContainer.test.tsx:10]: Just adding the type here (as in other files) and adding the required fields we were missing.

File: dashboard/src/containers/RoutesContainer/Routes.tsx [hunk 6]
```diff
@@ -81,10 +81,14 @@ class Routes extends React.Component<IRoutesProps> {
...
-      return <Redirect to={app.apps.list(this.props.cluster, this.props.currentNamespace)} />;
+      return (
+        <Redirect
+          to={{ pathname: app.apps.list(this.props.cluster, this.props.currentNamespace) }}
+        />
+      );
     }
     // There is not a default namespace, redirect to login page
-    return <Redirect to={"/login"} />;
+    return <Redirect to={{ pathname: "/login" }} />;
```
Comment (antgamdia) [inline dashboard/src/containers/RoutesContainer/Routes.tsx:91]: After upgrading the types, `to` has a required `pathname` property we were missing. I've updated the tests accordingly.

File: dashboard/package.json [hunk 7]
```diff
@@ -37,6 +19,7 @@
     "google-protobuf": "^3.17.3",
     "history": "^4.10.1",
     "immutable": "^4.0.0-rc.14",
+    "isomorphic-form-data": "^2.0.0",
```
Comment (absoludity) [inline dashboard/package.json:22]: So it sounds from @dlaloue-vmware 's comments that this dependency is required by swagger-ui-react but wasn't declared as a dependency by them? Or something similar: https://github.com/swagger-api/swagger-ui/issues/7436

Hah, although they mention there that: "On Thursday 29.7.2021 there will a new release of swagger-ui-react that will fix this issue automatically by updating to it.". Not sure if we wait, but if not, let's at least include a #TODO to remove it refering to the swagger-ui issue ?

**Review by absoludity — APPROVED** [review]
Great - one small comment about the new dep (which may not be required soon).

File: dashboard/package.json (hunk 7 above)
Comment (antgamdia) [reply dashboard/package.json:22]: While writing the message, they have just bumped up the version: https://github.com/swagger-api/swagger-ui/releases/tag/v3.51.2

**Pull request merged**